modules/service/src/test/scala/ch/epfl/bluebrain/nexus/kg/service/queue/KafkaConsumerSpec.scala

```diff
@@ -0,0 +1,74 @@
+
+class KafkaConsumerSpec extends WordSpecLike with Eventually with Matchers with EmbeddedKafka {
+
+  private implicit val as: ActorSystem              = ActorSystem("embedded-kafka")
+  private implicit val ec: ExecutionContextExecutor = as.dispatcher
+  private implicit val eventDecoder: Decoder[Event] = EventJsonLdSerialization.eventDecoder
+
+  private val counter          = new AtomicInteger
+  private val consumerSettings = ConsumerSettings(as, new StringDeserializer, new StringDeserializer)
+  private val path             = "a" / "b" / "c"
+  private val instant          = Instant.ofEpochMilli(1)
+  private val anonymous        = Anonymous(IdentityId("http://localhost/prefix/anonymous"))
+  private val alice            = UserRef(IdentityId("http://localhost/prefix/realms/realm/users/alice"))
+  private val meta             = Meta(alice, instant)
+  private val ownReadWrite     = Permissions(Permission.Own, Permission.Read, Permission.Write)
+  private val acl              = AccessControlList(anonymous -> ownReadWrite)
+  private val messages = List(
+    """{"@context":"http://localhost/prefix/context","path":"a/b/c","identity":{"@id":"http://localhost/prefix/realms/realm/groups/some-group","realm":"realm","group":"some-group","@type":"GroupRef"},"permissions":["own","read","write"],"meta":{"author":{"@id":"http://localhost/prefix/realms/realm/users/alice","realm":"realm","sub":"alice","@type":"UserRef"},"instant":"1970-01-01T00:00:00.001Z"},"@type":"PermissionsAdded"}""",
+    """{"@context":"http://localhost/prefix/context","path":"a/b/c","meta":{"author":{"@id":"http://localhost/prefix/realms/realm/users/alice","realm":"realm","sub":"alice","@type":"UserRef"},"instant":"1970-01-01T00:00:00.001Z"},"@type":"PermissionsCleared"}""",
+    """{"@context":"http://localhost/prefix/context","path":"a/b/c","acl":[{"identity":{"@id":"http://localhost/prefix/anonymous","@type":"Anonymous"},"permissions":["own","read","write"]}],"meta":{"author":{"@id":"http://localhost/prefix/realms/realm/users/alice","realm":"realm","sub":"alice","@type":"UserRef"},"instant":"1970-01-01T00:00:00.001Z"},"@type":"PermissionsCreated"}""",
+    """{"@context":"http://localhost/prefix/context","path":"a/b/c","identity":{"@id":"http://localhost/prefix/realms/realm/authenticated","realm":"realm","@type":"AuthenticatedRef"},"meta":{"author":{"@id":"http://localhost/prefix/realms/realm/users/alice","realm":"realm","sub":"alice","@type":"UserRef"},"instant":"1970-01-01T00:00:00.001Z"},"@type":"PermissionsRemoved"}""",
+    """{"@context":"http://localhost/prefix/context","path":"a/b/c","identity":{"@id":"http://localhost/prefix/realms/realm/users/alice","realm":"realm","sub":"alice","@type":"UserRef"},"permissions":["own","read","write"],"meta":{"author":{"@id":"http://localhost/prefix/realms/realm/users/alice","realm":"realm","sub":"alice","@type":"UserRef"},"instant":"1970-01-01T00:00:00.001Z"},"@type":"PermissionsSubtracted"}"""
+  )
+
+  "KafkaConsumer" should {
```

Comment:
We will need to add a test to check the offset, meaning: stopping the consumer and restarting it should only consume from the offset where it finished and it shouldn't re-consume events.